compiler/ml/printtyp.ml

```diff
@@ -52,24 +52,32 @@ let ident ppf id = pp_print_string ppf (ident_name id)
 (* Print a path *)
 
 let ident_pervasives = Ident.create_persistent "Pervasives"
+let ident_stdlib = Ident.create_persistent "Stdlib"
 let printing_env = ref Env.empty
-let non_shadowed_pervasive = function
+let non_shadowed_pervasive_or_stdlib = function
   | Pdot (Pident id, s, _pos) as path -> (
-    Ident.same id ident_pervasives
+    (Ident.same id ident_pervasives || Ident.same id ident_stdlib)
     &&
     try Path.same path (Env.lookup_type (Lident s) !printing_env)
     with Not_found -> true)
   | _ -> false
 
 let rec tree_of_path = function
   | Pident id -> Oide_ident (ident_name id)
-  | Pdot (_, s, _pos) as path when non_shadowed_pervasive path -> Oide_ident s
+  | Pdot (_, s, _pos) as path when non_shadowed_pervasive_or_stdlib path ->
+    Oide_ident s
+  | Pdot (p, s, _pos) when String.starts_with (Path.name p) ~prefix:"Stdlib_" ->
+    let path_name = Path.name p in
+    let ident_without_stdlib_prefix =
+      String.sub path_name 7 (String.length path_name - 7)
+    in
+    Oide_dot (Oide_ident ident_without_stdlib_prefix, s)
```

Comment:
I guess we could list all `Stdlib_*` with proper idents here if we wanted..? Instead of doing a string match.

There's also the question of whether stdlib types should be dug to when possible (`Date.t -> date`), but that's probably a separate question from this PR.

Reply:
And especially `Promise.t` -> `promise` as that's a built-in type.

Reply:
Yeah, that's a much better example, especially since `date` does not exist 🤦 😄